stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -2350,7 +2350,7 @@ function Base.fill!(V::SubArray{Tv, <:Any, <:SparseMatrixCSC, Tuple{Vararg{Union
     if (I[1] < 1 || I[end] > A.m) || (J[1] < 1 || J[end] > A.n)
         throw(BoundsError(A, (I, J)))
     end
-    if x == 0
+    if iszero(x)
```

Comment:
This will throw an error for types which don't support `iszero`, for example `String` (this has been discussed elsewhere IIRC). I think `iszero` should only be used when the argument is guaranteed to be a number.

Same below.